File: common/src/main/java/cn/breadnicecat/candycraftce/gui/block/menus/slot/ResultSlot.java
package cn.breadnicecat.candycraftce.gui.block.menus.slot;

import net.minecraft.world.Container;
import net.minecraft.world.inventory.Slot;
import net.minecraft.world.item.ItemStack;
import org.jetbrains.annotations.NotNull;

/**
 * @author <a href="https://gitee.com/Bread_NiceCat">Bread_NiceCat</a>
 * @date 2023/1/23 17:05
 */
public class ResultSlot extends Slot {
	public ResultSlot(Container pContainer, int pSlot, int pX, int pY) {
		super(pContainer, pSlot, pX, pY);
	}

	@Override
	public boolean mayPlace(@NotNull ItemStack pStack) {
		return false;
	}
}

File: src/main/kotlin/cn/breadnicecat/candycraftce/core/gui/block/menus/slots/ResultSlot.kt
package cn.breadnicecat.candycraftce.core.gui.block.menus.slots

import net.minecraft.world.Container
import net.minecraft.world.inventory.Slot
import net.minecraft.world.item.ItemStack

/**
 * 原版[net.minecraft.world.inventory.ResultSlot]过于繁琐，这里写一个简单化的版本
 *
 * @author [Bread_NiceCat](https://gitee.com/Bread_NiceCat)
 * @date 2023/1/23 17:05
 */
open class ResultSlot(pContainer: Container, pSlot: Int, pX: Int, pY: Int) : Slot(pContainer, pSlot, pX, pY) {
    override fun mayPlace(pStack: ItemStack): Boolean {
        return false
    }
}
